lib/gui/models/store.js

```diff
@@ -451,35 +451,37 @@ const storeReducer = (state = DEFAULT_STATE, action) => {
       return state.deleteIn([ 'selection', 'image' ]);
     }
 
-    case ACTIONS.SET_SETTING: {
-      const key = action.data.key;
-      const value = action.data.value;
-
-      if (!key) {
+    case ACTIONS.SET_SETTINGS: {
+      if (!action.data) {
         throw errors.createError({
-          title: 'Missing setting key'
+          title: 'Missing settings'
         });
       }
 
-      if (!_.isString(key)) {
+      if (!_.isPlainObject(action.data)) {
         throw errors.createError({
-          title: `Invalid setting key: ${key}`
+          title: `Invalid settings: ${action.data}`
         });
       }
 
-      if (!DEFAULT_STATE.get('settings').has(key)) {
+      const invalidKey = _.find(_.keys(action.data), (key) => {
+        return !_.isString(key);
+      });
+
+      if (!_.isNil(invalidKey)) {
         throw errors.createError({
-          title: `Unsupported setting: ${key}`
+          title: `Invalid setting key: ${invalidKey}`
         });
       }
 
-      if (_.isObject(value)) {
+      const invalidValue = _.find(_.values(action.data), _.overSome([ _.isArray, _.isObject ]));
```

Comment:
Why did it previously only check `_.isObject` but now it's checking both `_.isArray` and `_.isObject` ?